core/src/main/java/org/apache/cxf/validation/AbstractValidationInterceptor.java

```diff
@@ -108,15 +107,7 @@ protected Method getServiceMethod(Message message) {
         Message inMessage = message.getExchange().getInMessage();
         Method method = null;
         if (inMessage != null) {
-            method = (Method)inMessage.get("org.apache.cxf.resource.method");
-            if (method == null) {
-                BindingOperationInfo bop = inMessage.getExchange().getBindingOperationInfo();
-                if (bop != null) {
-                    MethodDispatcher md = (MethodDispatcher)
-                        inMessage.getExchange().getService().get(MethodDispatcher.class.getName());
-                    method = md.getMethod(bop);
-                }
-            }
+            method = MessageUtils.getTargetMethod(inMessage, null);
```

Comment:
Minor, may be introducing the overloaded method `MessageUtils.getTargetMethod(inMessage)` (which calls the original one with `null` supplier) would be a bit more cleaner, no need to guess what second argument is. Alternatively, could be `Optional<Method>` so the caller can raise exceptions or get along with `null`. What do you think?

Reply:
Good idea!  I'll give the `Optional<Method>` approach a try.  Thanks!